cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java

```diff
@@ -107,9 +108,24 @@ public StreamFetchHandler(CConfiguration cConf, Authenticator authenticator,
   public void fetch(HttpRequest request, HttpResponder responder,
                     @PathParam("namespace-id") String namespaceId,
                     @PathParam("stream") String stream,
-                    @QueryParam("start") long startTime,
-                    @QueryParam("end") @DefaultValue("9223372036854775807") long endTime,
+                    @QueryParam("start") String start,
+                    @QueryParam("end") @DefaultValue("9223372036854775807") String end,
                     @QueryParam("limit") @DefaultValue("2147483647") int limit) throws Exception {
+    long startTime;
+    long endTime;
+    try {
+      /* if string is numeric then value is in milliseconds */
+      startTime = Long.parseLong(start);
+    } catch (NumberFormatException nfe) {
+      /* else TimeMathParser returns a value in seconds and we must convert to milliseconds */
+      startTime = TimeUnit.SECONDS.toMillis(TimeMathParser.parseTime(start));
```

Comment:
I think it's better to have this `try-catch` logic as part of `TimeMathParser.parseTime`. Also, it's better to have a overload one that takes a `TimeUnit` so that the return time is in that given unit (that is Java idiomatic way as well).

`long startTime = TimeMathParser.parseTime(start, TimeUnit.MILLISECONDS);`